### PR TITLE
iOS: Added "run-on-main-thread" flag to tiapp.xml

### PR DESCRIPTION
**Pull request by pec1985**
This is a follow up from PR #6891 - changes that I forgot to commit

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ function iOSBuilder() {
 
 	// when true, uses the JavaScriptCore that ships with iOS instead of the original Titanium version
 	this.useJSCore = false;
-
+	this.runOnMainThread = true;
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Add whitespace before and after this line. Put a comment on the line above it to describe what this flag does.

Should this default to `true`?

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -2015,6 +2015,10 @@ iOSBuilder.prototype.initialize = function initialize() {
 	// TIMOB-17892
 	this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && this.cli.tiapp.ios && this.cli.tiapp.ios['use-jscore-framework'];
 
+	if (this.cli.tiapp.ios && this.cli.tiapp.ios['run-on-main-thread'] !== undefined) {
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: If you're setting anything on `this.currentBuildManifest`, you must not do it conditionally. Change this whole block to the following:

```
this.currentBuildManifest.runOnMainThread = this.runOnMainThread = this.tiapp.ios && this.tiapp.ios['run-on-main-thread'];
```

File: iphone/cli/commands/_build.js [hunk 3]
```diff
@@ -2286,6 +2290,14 @@ iOSBuilder.prototype.checkIfNeedToRecompile = function checkIfNeedToRecompile()
 			return true;
 		}
 
+		// check if the use RunOnMainThread flag has changed
+		if (this.runOnMainThread !== manifest.runOnMainThread) {
+			this.logger.info(__('Forcing rebuild: use RunOnMainThread flag changed since last build'));
+			this.logger.info('  ' + __('Was: %s', manifest.useJSCore));
+			this.logger.info('  ' + __('Now: %s', this.useJSCore));
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Should be `manifest.runOnMainThread` and `this.runOnMainThread`, not `manifest.useJSCore` and `this.useJSCore`.

File: iphone/cli/commands/_build.js [hunk 4]
```diff
@@ -4229,15 +4241,24 @@ iOSBuilder.prototype.processTiSymbols = function processTiSymbols() {
 	if (this.target === 'simulator' || this.includeAllTiModules) {
 		var definesFile = path.join(this.platformPath, 'Classes', 'defines.h');
 
+		contents = fs.readFileSync(definesFile).toString();
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: The file is now being read in twice. You only need to read in the file if `!this.runOnMainThread || this.useJSCore`. Please add a if block around this and conditionally check each specific flag like this:

```
var definesFile = path.join(this.platformPath, 'Classes', 'defines.h');

if (this.runOnMainThread && !this.useJSCore) {
	// just symlink the file
	if (!this.copyFileSync(definesFile, dest)) {
		this.logger.trace(__('No change, skipping %s', dest.cyan));
	}
	return;
}

contents = fs.readFileSync(definesFile).toString();
if (this.useJSCore) {
	contents += '\n#define USE_JSCORE_FRAMEWORK';
}
if (!this.runOnMainThread) {
	contents += '\n#define TI_USE_KROLL_THREAD';
}
```

File: iphone/cli/commands/_build.js [hunk 5]
```diff
@@ -4229,15 +4241,24 @@ iOSBuilder.prototype.processTiSymbols = function processTiSymbols() {
...
+			hasChanged = true;
+		}
+		if (!hasChanged) {
 			// just symlink the file
 			if (!this.copyFileSync(definesFile, dest)) {
 				this.logger.trace(__('No change, skipping %s', dest.cyan));
 			}
 			return;
 		}
+		contents += '\n';
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: I don't think we need this.

File: iphone/cli/commands/_build.js [hunk 6]
```diff
@@ -4273,7 +4294,9 @@ iOSBuilder.prototype.processTiSymbols = function processTiSymbols() {
 		if (this.useJSCore) {
 			contents.push('#define USE_JSCORE_FRAMEWORK')
 		}
-
+		if (!this.runOnMainThread) {
+			content.push('\n#define TI_USE_KROLL_THREAD');
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: I don't think you need the `\n`.

**Comment by cb1kenobi** [comment]
What is the associated ticket?

**Comment by cb1kenobi** [comment]
BTW, your changes to tiappxml.js look good!

File: iphone/cli/commands/_build.js (hunk 1 above)
Comment (pec1985) [inline iphone/cli/commands/_build.js]: Yes, defaults to `true`, I'll fix the white spaces and comments issue

File: iphone/cli/commands/_build.js (hunk 3 above)
Comment (pec1985) [inline iphone/cli/commands/_build.js]: Oops, good catch

File: iphone/cli/commands/_build.js (hunk 5 above)
Comment (pec1985) [inline iphone/cli/commands/_build.js]: I like to end a file with an empty line, that's all

**Comment by pec1985** [comment]
Thanks Chris, I'll fix the issues. The associated (old) ticket is TIMOB-18976

**Comment by cb1kenobi** [comment]
Looks good. CR'd and FR'd. **APPROVED**

**Pull request merged**